docs/developers/Backend.md

```diff
@@ -0,0 +1,125 @@
+* Call: An instantiated request in Cromwell of a thing to run. To further the function analogy this would be an
+  invocation of that function.
+* Job: The physical manifestation of a call by the backend. Examples of this would be an SGE job, a unix process, or a
+  Google Life Sciences operation ID.
+
+### Backend Lifecycle
+
+Within a running workflow, the backend is used in the following manner:
+
+* Initialization: Initialization routine called the first time a workflow uses a backend
+* Execute: The workflow requests that the backend run a job
+* Recover: Attempt to reconnect to a previously started job. An example of this would be if Cromwell was restarted and
+  wanted to reattach to currently running jobs.
+* Abort: Request that a running job be halted
+* Finalization: When a workflow is complete, allows for any workflow level cleanup to take place.
+
+The initialization and finalization steps are optional and are provided for cases where a backend needs that behavior;
+not all backends will need these.
+
+The implementations of both the recover and abort steps are up to the backend developer. For instance the recover
+function could be implemented to actually perform an execution and/or the abort function could be written to do nothing
+at all. Implementing these in a more robust manner is recommended for most cases but neither are universally
+appropriate.
+
+### How do I create a Backend?
+
+This section is assuming that you both know the underlying execution platform you wish to use and that you know how to
+programmatically submit work to it.
+
+The Cromwell engine uses three different types of Akka actors in the backend while processing a workflow:
+
+* `BackendWorkflowInitializationActor`: Handles the initialization phase
+* `BackendJobExecutionActor`: Handles requests to execute a new job or recover an existing job
+* `BackendWorkflowFinalizationActor`: Handles the finalization phase
+
+These three actors are all represented by a trait which a backend developer would need to implement. Both the
+initialization and finalization actors are optional.
+
+The following explanations of the traits mention multiple types defined in the Cromwell codebase. In particular
+`BackendJobDescriptor` wraps all the information necessary for a backend to instantiate a job and `JobKey` provides the
+information to uniquely identify a job. It is recommended that one look in the Cromwell codebase for more information on
+these and other types.
+
+#### BackendWorkflowInitializationActor
+
+If a backend developer wishes to take advantage of the initialization phase of the backend lifecycle they must implement
+this trait. There are three functions which must be implemented:
+
+1. `abortInitialization: Future[WorkflowAbortResponse]` specifies what to do, if anything, when a workflow is requested
+   to abort while a backend initialization is in progress.
```

Comment:
`BackendWorkflowInitializationActor.scala` says:
```
  def abortInitialization(): Unit = {}
```
I don't see any backends that override this, and it looks like we don't do anything with the result when we call it.

Which one do you think we should change?